src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -640,6 +701,56 @@ impl<'tcx> Tree {
         interp_ok(())
     }
 
+    /// Restores the "children are stronger" invariant of `latest_foreign_access` after a retag.
+    /// See also `reset_last_foreign_access_after_reborrow`.
+    /// Retags can violate the invariant on `latest_foreign_access`,
+    /// which states that a node's children survive accesses at least as strong as that node itself.
+    /// While retags act as a read for the offsets specified in the retag, it is also present at the other
```

Comment:
This same comment is copied 3 times it seems... seems better to instead reference a common pace where this is explained once.

Reply:
Indeed. But unfortunately `Option<AccessKind>` has no obvious place where one would add such an annotation.
Note that `Option<AccessKind>` already has the here-suggested order "natively" since it implements `Ord`, and it turns out that the auto-derived order is the correct one here (this is not entirely accidental). But relying on that seems wrong, and the code does *not* do that currently. Perhaps one could introduce a newtype?

Reply:
> But relying on that seems wrong, 

Why? Your comments treat the type as having an order. And it does even actually have that order. We should do that.

If you feel uneasy relying on it, add a little `#[test]` function asserting that the order is the way we want it to be.